indico/web/forms/fields.py

```diff
@@ -540,3 +541,26 @@ def __init__(self, *args, **kwargs):
 
     def _value(self):
         return self.text_value
+
+
+class IndicoColorSwitchField(JSONField):
+    """Field allowing user to pick a color from a set of predefined values"""
+
+    widget = JinjaWidget('forms/colorswitch_widget.html')
+    default_widget_colors = {'background': 'EEE0EF', 'text': '1D041F'}
+
+    def __init__(self, *args, **kwargs):
+        self.default_colors = kwargs.pop('default_colors', self.default_widget_colors)
+        super(IndicoColorSwitchField, self).__init__(*args, **kwargs)
+
+    def process_formdata(self, valuelist):
+        super(IndicoColorSwitchField, self).process_formdata(valuelist)
+        self.data = ColorTuple(self.data['text'], self.data['background'])
+
+    def populate_obj(self, obj, name):
+        setattr(obj, name, self.data)
+
+    def _value(self):
+        if not self.data:
+            return self.default_widget_colors
+        return {'background': self.data.background, 'text': self.data.text}
```

Comment:
Why not return a `ColorTuple` here, too? You can convert it to a dict with `._asdict()` (which I'd do in the template).